test_conformance/basic/test_progvar.cpp

```diff
@@ -547,12 +549,28 @@ static cl_int print_build_log(cl_program program, cl_uint num_devices,
     return error;
 }
 
-static void l_load_abilities(cl_device_id device)
+static cl_int l_load_abilities(cl_device_id device)
 {
+    cl_int status = CL_SUCCESS;
+
     l_has_half = is_extension_available(device, "cl_khr_fp16");
     l_has_double = is_extension_available(device, "cl_khr_fp64");
     l_has_cles_int64 = is_extension_available(device, "cles_khr_int64");
 
+    if (get_device_cl_version(device) >= Version(3, 0)) {
+        OpenCLCFeatures features;
+        status = get_device_cl_c_features(device, features);
+        if (status)
+        {
+            log_error("Couldn't query OpenCL C features for the device!\n");
+            return TEST_FAIL;
+        }
+
+        l_has_atomics =
+           features.supports__opencl_c_atomic_order_seq_cst
+           && features.supports__opencl_c_atomic_scope_device;
```

Comment:
This may be correct, but this will essentially mean that the atomic tests wont run on devices supporting OpenCL 1.2-level atomics.  Can this check be weakened at all?

Reply:
this would require rewriting the OpenCL code as it's using the optional `atomic_load` and `atomic_store` functions.

It could be done, just need to figure out if the kernels needs to be rewritten as we can't do device wide atomic operations without optional features.

Maybe the code is even fine with the scope being reduced to `memory_scope_work_group`. Would have to dig deeper into how it all works.

Reply:
Ah, gotcha.  I did a quick test and it does indeed appear that the kernels are using the `atomic_load` and `atomic_store` overloads with no additional arguments, so they do currently require both the `seq_cst` and `scope_device` features.  I do think these could be weakened to use a narrower scope or memory order, but that's beyond the scope of this PR.

Worth mentioning also, there is still pretty good testing coverage even without atomics - I count 141 total kernels with atomics, and 123 total kernels without atomics.

I'll just resolve this comment.  Thanks!